### PR TITLE
own: fetch user external accounts in one query.

### PR DESCRIPTION
**Pull request by sashaostrikov**
This commits adds `ListForUsers` function to `userExternalAccountsStore` which fetches all user external accounts for a slice of users, which allows us to use 1 SQL query instead of N in bag, while resolving user references.

Test plan:
Unit tests added/updated.

Closes https://github.com/sourcegraph/sourcegraph/issues/53862

### CODE REVIEW

**Comment by cbart** [comment]
I'll sit down to review the bag changes later today as this needs more care. Apologies that I didn't get to it earlier.

File: enterprise/internal/own/ownref.go [hunk 1]
```diff
@@ -351,32 +362,52 @@ func (r *userReferences) appendErr(err error) {
...
+// augment fetches all the references that are missing for all users in a batch.
+// These can then be linked back into the bag using `linkBack`. In order to call
+// augment, `id`.
+func (b userReferencesBatch) augment(ctx context.Context, db edb.EnterpriseDB) {
+	userIDsToFetchHandles := collections.NewSet[int32]()
+	for _, r := range b {
+		// User references has to have an ID.
+		if r.id == 0 {
+			r.appendErr(errors.New("userReferences needs id set for augmenting"))
+			return
```
Comment (cbart) [inline enterprise/internal/own/ownref.go:376]:
```suggestion
			continue
```

File: internal/database/external_accounts.go [hunk 2]
```diff
@@ -441,6 +444,40 @@ func (s *userExternalAccountsStore) List(ctx context.Context, opt ExternalAccoun
 	return s.listBySQL(ctx, sqlf.Sprintf("WHERE %s ORDER BY id ASC %s", sqlf.Join(conds, "AND"), opt.LimitOffset.SQL()))
 }
 
+func (s *userExternalAccountsStore) ListForUsers(ctx context.Context, userIDs []int32) (userToAccts map[int32][]*extsvc.Account, err error) {
+	tr, ctx := trace.New(ctx, "UserExternalAccountsStore.ListForUsers", "")
+	var count int
```
Comment (cbart) [inline internal/database/external_accounts.go:449]: is `count` used at all; seems like it's only assigned to?

**Review by cbart — APPROVED** [review]
:shipit:

File: internal/database/external_accounts.go (hunk 2 above)
Comment (sashaostrikov) [reply internal/database/external_accounts.go:449]: yep, for tracing on L457

File: enterprise/internal/own/ownref.go (hunk 1 above)
Comment (sashaostrikov) [reply enterprise/internal/own/ownref.go:376]: whoa, that's a great catch!

**Pull request merged**